collect_app/src/main/res/layout/form_entry_end.xml

```diff
@@ -77,46 +77,48 @@ the specific language governing permissions and limitations under the License.
 
-        <androidx.constraintlayout.widget.ConstraintLayout
+        <LinearLayout
             android:id="@+id/buttons_container"
-            android:layout_width="0dp"
+            android:layout_width="match_parent"
             android:layout_height="wrap_content"
             android:layout_marginTop="64dp"
-            app:layout_constraintWidth_max="640dp"
             app:layout_constraintBottom_toBottomOf="parent"
             app:layout_constraintEnd_toEndOf="parent"
             app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@+id/form_edits_warning">
+            app:layout_constraintTop_toBottomOf="@+id/form_edits_warning"
+            app:layout_constraintWidth_max="640dp">
 
             <com.google.android.material.button.MaterialButton
                 android:id="@+id/save_as_draft"
                 style="?materialButtonOutlinedIconStyle"
                 android:layout_width="0dp"
-                android:layout_height="wrap_content"
-                android:text="@string/save_as_draft"
+                android:layout_height="match_parent"
                 android:layout_marginEnd="@dimen/margin_extra_small"
-                app:layout_goneMarginEnd="0dp"
+                android:layout_weight="1"
+                android:text="@string/save_as_draft"
+                app:icon="@drawable/ic_save_menu_24"
+                app:iconGravity="textStart"
                 app:layout_constraintBottom_toBottomOf="parent"
                 app:layout_constraintEnd_toStartOf="@id/finalize"
                 app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toTopOf="parent"
-                app:icon="@drawable/ic_save_menu_24"
-                app:iconGravity="textStart" />
+                app:layout_goneMarginEnd="0dp" />
 
             <com.google.android.material.button.MaterialButton
                 android:id="@+id/finalize"
                 style="?materialButtonIconStyle"
                 android:layout_width="0dp"
-                android:layout_height="wrap_content"
-                android:text="@string/finalize"
+                android:layout_height="match_parent"
                 android:layout_marginStart="@dimen/margin_extra_small"
-                app:layout_goneMarginStart="0dp"
+                android:layout_weight="1"
+                android:text="@string/finalize"
+                app:icon="@drawable/ic_send_24"
+                app:iconGravity="textStart"
                 app:layout_constraintBottom_toBottomOf="parent"
                 app:layout_constraintEnd_toEndOf="parent"
                 app:layout_constraintStart_toEndOf="@id/save_as_draft"
                 app:layout_constraintTop_toTopOf="parent"
-                app:icon="@drawable/ic_send_24"
-                app:iconGravity="textStart" />
-        </androidx.constraintlayout.widget.ConstraintLayout>
+                app:layout_goneMarginStart="0dp" />
```

Comment:
I think this might not work as expected now as it's no longer `ConstraintLayout`. So if the first button is gone (because of the settings) there will be a margin visible.

Reply:
Ach! So annoying that we can't have two views match heights like this in `ConstraintLayout` I looked into it again and it seems like there still isn't a way (although so SOs have out of date answers). I've used `LinearLayout`'s dividers to achieve the same thing as the `goneMargin` attributes.